### ISSUE TITLE
Exsentry crashing when sending results to Sentry

### ISSUE THREAD

**Issue by ASCrookes** [appcues/exsentry#7]
I added in exsentry and everything seemed fine. If I make a call to my server (added a route that crashes to test) that throws an exception exsentry also throws an except. After too many times exsentry goes down and takes my app down with it. Any thoughts on what I'm doing wrong or if this is a bug with exsentry itself? I added the relevant logs, not sure if that'd help out at all.

crash.log
```
2016-03-31 19:58:33 =CRASH REPORT====
  crasher:
    initial call: Elixir.ExSentry.Sender:init/1
    pid: <0.441.0>
    registered_name: []
    exception exit: {{{badmap,[{'Cache-Control',<<"max-age=0">>},{'Connection',<<"close">>},{'Content-Language',<<"en">>},{'Content-Type',<<"application/json">>},{'Date',<<"Thu, 31 Mar 2016 19:57:50 GMT">>},{'Expires',<<"Thu, 31 Mar 2016 19:57:50 GMT">>},{'Last-Modified',<<"Thu, 31 Mar 2016 19:57:50 GMT">>},{'Server',<<"nginx">>},{'Transfer-Encoding',<<"chunked">>},{'Vary',[<<"Accept-Language, Cookie">>,<<"Accept-Encoding">>]},{'X-Content-Type-Options',<<"nosniff">>},{'X-Frame-Options',<<"SAMEORIGIN">>},{'X-Served-By',<<"web-5">>},{'X-XSS-Protection',<<"1; mode=block">>}]},[{maps,find,['X-Sentry-Error',[{'Cache-Control',<<"max-age=0">>},{'Connection',<<"close">>},{'Content-Language',<<"en">>},{'Content-Type',<<"application/json">>},{'Date',<<"Thu, 31 Mar 2016 19:57:50 GMT">>},{'Expires',<<"Thu, 31 Mar 2016 19:57:50 GMT">>},{'Last-Modified',<<"Thu, 31 Mar 2016 19:57:50 GMT">>},{'Server',<<"nginx">>},{'Transfer-Encoding',<<"chunked">>},{'Vary',[<<"Accept-Language, Cookie">>,<<"Accept-Encoding">>]},{'X-Content-Type-Options',<<"nosniff">>},{'X-Frame-Options',<<"SAMEORIGIN">>},{'X-Served-By',<<"web-5">>},{'X-XSS-Protection',<<"1; mode=block">>}]],[]},{'Elixir.Map',get,3,[{file,"lib/map.ex"},{line,131}]},{'Elixir.ExSentry.Sender',handle_cast,2,[{file,"lib/exsentry/sender.ex"},{line,83}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,615}]},{gen_server,handle_msg,5,[{file,"gen_server.erl"},{line,681}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]},[{gen_server,terminate,7,[{file,"gen_server.erl"},{line,826}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}
    ancestors: [exsentry_default_client,'Elixir.ExSentry.Supervisor',<0.224.0>]
    messages: []
    links: [<0.438.0>]
    dictionary: [{random_seed,{27839,21123,25074}}]
    trap_exit: false
    status: running
    heap_size: 4185
    stack_size: 27
    reductions: 7579
  neighbours:
    neighbour: [{pid,<0.438.0>},{registered_name,exsentry_default_client},{initial_call,{'Elixir.ExSentry.Client',init,['Argument__1']}},{current_function,{gen_server,loop,6}},{ancestors,['Elixir.ExSentry.Supervisor',<0.224.0>]},{messages,[]},{links,[<0.225.0>,<0.441.0>]},{dictionary,[]},{trap_exit,false},{status,waiting},{heap_size,28690},{stack_size,9},{reductions,7579}]
```

also crash.log
```
2016-03-31 19:58:33 =SUPERVISOR REPORT====
     Supervisor: {local,'Elixir.ExSentry.Supervisor'}
     Context:    shutdown
     Reason:     reached_max_restart_intensity
     Offender:   [{pid,<0.438.0>},{id,'Elixir.ExSentry.Client'},{mfargs,{'Elixir.ExSentry.Client',start_link,[[{name,exsentry_default_client}]]}},{restart_type,permanent},{shutdown,5000},{child_type,worker}]
```

**Comment by ASCrookes** [comment]
This might also be helpful

erlang.log
```
[error] GenServer #PID<0.410.0> terminating
** (BadMapError) expected a map, got: ["Cache-Control": "max-age=0", Connection: "close", "Content-Language": "en", "Content-Type": "application/json", Date: "Thu, 31 Mar 2016 20:17:18 GMT", Expires: "Thu, 31 Mar 2016 20:17:18 GMT", "Last-Modified": "Thu, 31 Mar 2016 20:17:18 GMT", Server: "nginx", "Transfer-Encoding": "chunked", Vary: ["Accept-Language, Cookie", "Accept-Encoding"], "X-Content-Type-Options": "nosniff", "X-Frame-Options": "SAMEORIGIN", "X-Served-By": "web-15", "X-XSS-Protection": "1; mode=block"]
    (stdlib) :maps.find(:"X-Sentry-Error", ["Cache-Control": "max-age=0", Connection: "close", "Content-Language": "en", "Content-Type": "application/json", Date: "Thu, 31 Mar 2016 20:17:18 GMT", Expires: "Thu, 31 Mar 2016 20:17:18 GMT", "Last-Modified": "Thu, 31 Mar 2016 20:17:18 GMT", Server: "nginx", "Transfer-Encoding": "chunked", Vary: ["Accept-Language, Cookie", "Accept-Encoding"], "X-Content-Type-Options": "nosniff", "X-Frame-Options": "SAMEORIGIN", "X-Served-By": "web-15", "X-XSS-Protection": "1; mode=block"])
    (elixir) lib/map.ex:131: Map.get/3
    (exsentry) lib/exsentry/sender.ex:83: ExSentry.Sender.handle_cast/2
    (stdlib) gen_server.erl:615: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:681: :gen_server.handle_msg/5
    (stdlib) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Last message: {:"$gen_cast", {:send, "https://app.getsentry.com/api/72385/store/", [{"X-Sentry-Auth", "Sentry sentry_version=7, sentry_client=\"ExSentry/0.3.0\", sentry_timestamp=1459455481, sentry_key=62403280a3fc4df3b80e6d2121e637d3, sentry_secret=76b32edc0a914a97b391113b591cc38d"}, {"Content-Type", "application/json"}], %{event_id: "417915747d39473a91cd39184f6c5964", level: :error, logger: "ExSentry 0.3.0", message: "undefined function Cybal.__schema__/1 (module Cybal is not available)", modules: %{plug: "1.0.2", phoenix_ecto: "1.2.0", mimerl: "1.0.0", ssl_verify_hostname: "1.0.5", uuid: "1.1.3", ssl: "7.2", hashids: "2.0.0", ibrowse: "4.2.2", lager: "2.1.0", setup: "1.4", jsx: "2.8.0", exometer: "1.1-47-gfc36768", combine: "0.7.0", ecto: "1.0.4", elixometer: "1.0.0", postgrex: "0.9.1", ranch: "1.1.0", timex: "2.1.0", erlcloud: "0.13.0-19-ga782896", mixpanel: "0.0.1", exredis: "0.2.0", fuzzyurl: "0.8.1", oauth: "1.5.0", sasl: "2.6.1", inets: "6.1", stdlib: "2.7", cymbal: "0.0.1", hackney: "1.4.4", cowlib: "1.0.1", idna: "1.0.2", iex: "1.2.0", phoenix_html: "2.2.0", httpotion: "2.1.0", eredis: "1.0.8", lhttpc: "1.3.0", httpoison: "0.8.1", cowboy: "1.0.3", goldrush: "0.1.6", poolboy: "1.5.1", certifi: "0.1.1", ...}, platform: "other", release: "0.3.0", request: %ExSentry.Model.Request{cookies: %{"__cfduid" => "d0cc7dc7af979f783d3d0b260910738711457469252"}, data: "", env: %{remote_ip: "127.0.0.1"}, headers: %{"accept-encoding" => "gzip", "access-token" => "19a84731-d1b6-4b8c-9ed5-486b3b272ed5", "cf-connecting-ip" => "207.41.188.221", "cf-ipcountry" => "US", "cf-ray" => "28c67be8ea9d00ad-EWR", "cf-respect-strong-etag" => "0", "cf-visitor" => "{\"scheme\":\"https\"}", "connection" => "close", "content-type" => "application/json", "cookie" => "__cfduid=d0cc7dc7af979f783d3d0b260910738711457469252", "host" => "fawkes-staging.cymbal.fm", "user-agent" => "Paw/2.3.2 (Macintosh; OS X/10.11.1) GCDHTTPRequest", "x-forwarded-for" => "207.41.188.221, 173.245.52.224, 172.31.48.4", "x-forwarded-port" => "80", "x-forwarded-proto" => "http", "x-real-ip" => "172.31.48.4"}, method: "GET", query_string: "", url: "/api/v1/user/2/anthems"}, server_name: "ip-172-31-13-178", stacktrace: %ExSentry.Model.Stacktrace{frames: [%{filename: "web/controllers/v1/user_controller.ex", function: "cymbals/3", lineno: 236, module: CymbalApi.V1.UserController}, %{filename: "web/controllers/v1/user_controller.ex", function: "action/2", lineno: 1, module: CymbalApi.V1.UserController}, %{filename: "web/controllers/v1/user_controller.ex", function: "phoenix_controller_pipeline/2", lineno: 1, module: CymbalApi.V1.UserController}, %{filename: "lib/phoenix/router.ex", function: "dispatch/2", lineno: 255, module: CymbalApi.V1.UserRouter}, %{filename: "web/routers/v1/user_router.ex", function: "do_call/2", lineno: 1, module: CymbalApi.V1.UserRouter}, %{filename: "lib/phoenix/router/route.ex", function: "forward/4", lineno: 157, module: Phoenix.Router.Route}, %{filename: "lib/phoenix/router.ex", function: "dispatch/2", lineno: 255, module: Cymbal.Router}]}, timestamp: "2016-03-31T20:18:01"}, 0}}
State: %ExSentry.Sender.State{delay: 1000, logging: true, retries: 3, status: :launched, testing: false, timeout: 3000}
```

**Comment by ASCrookes** [comment]
Issue with different versions of HTTPotion. Mine was using keyword lists while the updated dependency uses maps.
